src/gurt/hash.c

```diff
@@ -1022,23 +1022,26 @@ link2rlink(d_list_t *link)
 static void
 rl_op_addref(struct d_rlink *rlink)
 {
-	rlink->rl_ref++;
+	atomic_fetch_add_relaxed(&rlink->rl_ref, 1);
 }
 
 static bool
 rl_op_decref(struct d_rlink *rlink)
 {
-	D_ASSERT(rlink->rl_ref > 0);
-	rlink->rl_ref--;
-	return rlink->rl_ref == 0;
+	uint32_t oldref;
+
+	oldref = atomic_fetch_sub_relaxed(&rlink->rl_ref, 1);
+	D_ASSERT(oldref > 0);
+
+	return oldref == 1;
 }
 
 static void
 rl_op_init(struct d_rlink *rlink)
 {
 	D_INIT_LIST_HEAD(&rlink->rl_link);
 	rlink->rl_initialized	= 1;
-	rlink->rl_ref		= 1; /* for caller */
+	atomic_store_relaxed(&rlink->rl_ref, 1); /* for caller */
```

Comment:
The assignments here were aligned so for correctness you should probably re-indent the lime above to leave only one space before the assignment.